src/neo/SmartContract/Helper.cs

```diff
@@ -162,7 +162,7 @@ internal static bool VerifyWitnesses(this IVerifiable verifiable, StoreView snap
                     if (hashes[i] != verifiable.Witnesses[i].ScriptHash) return false;
                     offset = 0;
                 }
-                using (ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, verifiable, snapshot, gas))
+                using (ApplicationEngine engine = ApplicationEngine.Create(TriggerType.Verification, verifiable, snapshot.Clone(), gas))
```

Comment:
Maybe we can `Clone` outside the for ?

Reply:
`Verification`s of different transactions should not be able to influence each other. Because transactions in different nodes have different orders. And the `Clone()` method doesn't cost much resources.

Reply:
But it's the same TX with different verification script, all of them should use the same snapshot, isn't it?

Reply:
I think they should be isolated from each other too.